src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private RandomSize size = new RandomSize();
+
+    private Figure circle = new Circle(size.getSize(), colorSupplier.getRandomColor());
+    private Figure isoscelesTrapezoid = new IsoscelesTrapezoid(size.getSize(),
+            size.getSize(), size.getSize(), colorSupplier.getRandomColor());
+    private Figure rectangle = new Rectangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());
+    private Figure square = new Circle(size.getSize(),
+            colorSupplier.getRandomColor());
+    private Figure rightTriangle = new RightTriangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());
+
+    private Figure [] figures = {
+            circle,
+            isoscelesTrapezoid,
+            rectangle,
+            square,
+            rightTriangle
+    };

```

Comment:
You should not store figures as an array at the class level for the purpose of random figure generation. Instead, create them on-the-fly inside the 'getRandomFigure()' method.